### DISCUSSION TITLE
Recommended Debugging Flow

### DISCUSSION THREAD

**Discussion by Cjewett** [dotnet/sdk-container-builds#417] [Q&A] [answered]
I'm taking a look at https://github.com/dotnet/sdk-container-builds/issues/146 and was curious if the owners or contributors for this repository have a better debug flow than building the package into a local NuGet feed, referencing it from another project, and running a build while outputting logs. I'm pretty much looking for a way to step into the code as it is happening but I'm struggling to connect the dots between debugging msbuild and debugging packages that add new tasks to a project. Any guidance would be much appreciated!

Tagging @baronfel as you seem to be active and also commented in the issue.

**Comment by rainersigwald** [comment, marked as answer]
One trick that is often useful is to set `MSBUILDDEBUGONSTART=1`, attach to MSBuild, and set a breakpoint at `{TaskName}.Execute`, for instance `CreateNewImage.Execute`.

**Reply by Cjewett** [reply]
Thanks! Between this and altering csproj with the necessary targets to inherit newly built local packages I got a setup going.
